### PR TITLE
Tweak user & org models, then remove unused user/admin endpoints

### PR DESCRIPTION
**Pull request by joel-wright**
# Description

Tweak user & org models to support multi-tenant admin authorisation & super users

## Type of change

Please select the option(s) below that are most relevant:

- [x] Cleanup
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

Existing tests updated

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/api/api_v1/routers/auth.py [hunk 1]
```diff
@@ -18,13 +19,15 @@ async def login(db=Depends(get_db), form_data: OAuth2PasswordRequestForm = Depen
         extra={"props": {"user_id": form_data.username}},
     )
 
-    user = authenticate_user(db, form_data.username, form_data.password)
-    if not user:
+    authenticated = authenticate_user(db, form_data.username, form_data.password)
```
Comment (THOR300) [inline app/api/api_v1/routers/auth.py:22]: Could just deconstruct the response to user, app_user here?

File: alembic/versions/0014_add_app_user_per_org_authz.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+
+# revision identifiers, used by Alembic.
+revision = '0014'
+down_revision = '0013'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
```
Comment (THOR300) [inline alembic/versions/0014_add_app_user_per_org_authz.py:19]: To confirm the understanding, we are creating a user table that all users will be listed in. We then add permissions for that user to have access to each organisation using the organisation_admin table with the primary key being the unique user email and org id?

File: app/core/auth.py [hunk 3]
```diff
@@ -18,82 +17,68 @@
...
             is_superuser=permissions == "admin",
             is_active=is_active,
+            authorization=authorization,
         )
         return jwt_user
     except PyJWTError:
         raise CREDENTIALS_EXCEPTION
 
 
-async def get_current_active_user(
```
Comment (THOR300) [inline app/core/auth.py:47]: Assuming converting from async as we don't expect to have many authentication requests at all as users aren't authenticating?

File: app/db/models/app/users.py [hunk 4]
```diff
@@ -26,13 +26,15 @@ class Organisation(Base):
     organisation_type = sa.Column(sa.String)
 
 
-class AdminOrganisation(Base):
+class OrganisationUser(Base):
```
Comment (THOR300) [inline app/db/models/app/users.py:29]: Arguably could be called OrganisationAdmin?

File: app/db/models/law_policy/geography.py [hunk 5]
```diff
@@ -11,7 +11,7 @@ class Geography(Base):  # noqa: D101
     id = sa.Column(sa.Integer, primary_key=True)
     # to display to end-users
     display_value = sa.Column(sa.Text, unique=True, nullable=False)
-    slug = sa.Column(sa.Text, unique=True, nullable=False, index=True)
+    slug = sa.Column(sa.Text, nullable=False, unique=True, index=True)
```
Comment (THOR300) [inline app/db/models/law_policy/geography.py:14]: Just looks like a pre-commit refactor?

File: tests/routes/test_admin.py [hunk 6]
```diff
@@ -24,262 +22,22 @@
     Instrument,
     Category,
     Keyword,
-    User,
-    PasswordResetToken,
 )
 from app.db.models.law_policy import Geography
 
 
-def test_get_users(client, test_superuser, superuser_token_headers):
```
Comment (THOR300) [inline tests/routes/test_admin.py:33]: Do we need to replace these tests with some basic tests to use the new models post db update?

**Review by THOR300 — APPROVED** [review]
Looks good just some minor queries from myself.

File: app/api/api_v1/routers/auth.py (hunk 1 above)
Comment (joel-wright) [reply app/api/api_v1/routers/auth.py:22]: authenticated could be `None` here

File: alembic/versions/0014_add_app_user_per_org_authz.py (hunk 2 above)
Comment (joel-wright) [reply alembic/versions/0014_add_app_user_per_org_authz.py:19]: yes, correct

File: app/core/auth.py (hunk 3 above)
Comment (joel-wright) [reply app/core/auth.py:47]: none of the code was actually async, or called in an async way, so I'm just cleaning up a potential issue here

File: app/db/models/app/users.py (hunk 4 above)
Comment (joel-wright) [reply app/db/models/app/users.py:29]: Honestly, this was just written with the expectation that we may want different roles in the future.

File: app/db/models/law_policy/geography.py (hunk 5 above)
Comment (joel-wright) [reply app/db/models/law_policy/geography.py:14]: yeah, I thought I'd undone a change - the alembic generated migration was wrong & I was trying to get it to generate something valid. In the end I just had to re-order some of the migration lines manually :(

File: tests/routes/test_admin.py (hunk 6 above)
Comment (joel-wright) [reply tests/routes/test_admin.py:33]: All this functionality has been removed - user management will be manual for a while, so there's no code to test. We will probably need these sorts of tests again, but that's a task for future-us

**Pull request merged**